### PR TITLE
1073 Containerizing demo-app

### PR DESCRIPTION
**Pull request by 3uzbcqje**
Containerizing demo-app

### CODE REVIEW

**Review by dysfunctor — APPROVED** [review]
LGTM

File: examples/demo-app/core/scripts/shell_setup.sh [hunk 1]
```diff
@@ -1,10 +1,15 @@
-# Add the `upa` and `demo-app` commands to the current shell.
-[ "${ZSH_VERSION}" = "" ] && this_dir=`dirname ${BASH_SOURCE[0]}` || this_dir=`dirname ${(%):-%N}`
-demo_app_root_dir=`realpath ${this_dir}/..`
+#!/usr/bin/env bash
+
+# https://stackoverflow.com/questions/59895/how-do-i-get-the-directory-where-a-bash-script-is-located-from-within-the-script
+export SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
```
Comment (dtebbs) [inline examples/demo-app/core/scripts/shell_setup.sh]: what's wrong with the original?  Doesn't this break zsh?

File: examples/demo-app/core/scripts/shell_setup.sh [hunk 2]
```diff
@@ -1,10 +1,15 @@
-# Add the `upa` and `demo-app` commands to the current shell.
-[ "${ZSH_VERSION}" = "" ] && this_dir=`dirname ${BASH_SOURCE[0]}` || this_dir=`dirname ${(%):-%N}`
-demo_app_root_dir=`realpath ${this_dir}/..`
+#!/usr/bin/env bash
```
Comment (dtebbs) [inline examples/demo-app/core/scripts/shell_setup.sh]: Thsi is intended to be sourced, no executed standlone

File: examples/demo-app/core/scripts/shell_setup.sh [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
+
+# https://stackoverflow.com/questions/59895/how-do-i-get-the-directory-where-a-bash-script-is-located-from-within-the-script
+export SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+export demo_app_root_dir=$(realpath ${SCRIPT_DIR}/..)
```
Comment (dtebbs) [inline examples/demo-app/core/scripts/shell_setup.sh]: why export here and above?  This is just for sourcing.

**Pull request merged**